lib/contracts/blockchain.js

```diff
@@ -256,45 +256,8 @@ class Blockchain {
         if (!self.contractsConfig || !self.contractsConfig.deployment || !self.contractsConfig.deployment.host) {
           return next();
         }
-        const origin = self.blockchainConfig.wsOrigins.split(',')[0];
-        const options = {
-          protocolVersion: 13,
-          perMessageDeflate: true,
-          origin: origin,
-          host: self.contractsConfig.deployment.host,
-          port: self.contractsConfig.deployment.port
-        };
-        if (self.contractsConfig.deployment.type === 'ws') {
-          options.headers = {
-            'Sec-WebSocket-Version': 13,
-            Connection: 'Upgrade',
-            Upgrade: 'websocket',
-            'Sec-WebSocket-Extensions': 'permessage-deflate; client_max_window_bits',
-            Origin: origin
-          };
-        }
-        let req;
-        // remove trailing api key from infura, ie rinkeby.infura.io/nmY8WtT4QfEwz2S7wTbl
-        if(options.host.indexOf('/') > -1){
-          options.host = options.host.split('/')[0];
-        }
-        if((self.contractsConfig.deployment.protocol || 'http') === 'https'){
-          req = require('https').get(options);          
-        }else{
-          req = require('http').get(options);
-        }
-        
-        req.on('error', (err) => {
-          next(err);
-        });
-
-        req.on('response', (_response) => {
-          next();
-        });
-
-        req.on('upgrade', (_res, _socket, _head) => {
-          next();
-        });
+        const {host, port, type, protocol}  = self.contractsConfig.deployment;
```

Comment:
Should we handle the case where protocol is not specified in the config, ie default to `http`? Or possibly amend the default contracts config options to include `"protocol": "http"`?